payjoin-relay/src/main.rs

```diff
@@ -307,40 +315,47 @@ async fn post_fallback(
         return Err(HandlerError::PayloadTooLarge);
     }
 
-    match pool.push_req(&id, req.into()).await {
+    match pool.lock().await.push_req(&id, req.to_vec()).await {
```

Comment:
exposing the Db locking to the application is a good sign your abstraction is leaking